### ISSUE TITLE
CraftVariable::EVENT_INIT event not working in modules since 3.0.5

### ISSUE THREAD

**Issue by Wiejeben** [craftcms/cms#2831]
### Description
After updating to 3.0.5, `\craft\web\twig\variables\CraftVariable::EVENT_INIT` stopped working inside my module, reverting back to 3.0.4 fixes this.

I have a feeling it might be broken because of this change:

https://github.com/craftcms/cms/commit/610f7ac8016a5adcb8606b831afdd6322da57dea#diff-d735858371e00b6108b199120931723f

### Steps to reproduce
1. Make sure you are running 3.0.5.
2. Add the following inside your (`__construct()` or) `init()` method inside your module that is being loaded on bootstrap:
```
// Register template variables
\yii\base\Event::on(
    \craft\web\twig\variables\CraftVariable::class,
    \craft\web\twig\variables\CraftVariable::EVENT_INIT,
    function(\yii\base\Event $event) {
        die('It won\'t');
    }
);
```
3. Nothing happens.

### Additional info

- Craft version: 3.0.5
- PHP version: 7.0.10

**Comment by brandonkelly** [comment]
Working on my end. Do you have any plugins installed?

**Comment by Wiejeben** [comment]
Ah right, I disabled all my plugins and noticed that fixed the error. Will be looking into the plugin that is causing the `CraftVariable::EVENT_INIT` to break.

**Comment by brandonkelly** [comment]
Most likely it’s because a plugin is actually loading Twig. If you can figure out which one, I can look into a better solution on my end.

**Comment by Wiejeben** [comment]
Right, I did some more debugging.

SEOmatic does some stuff right after `Plugins::EVENT_AFTER_LOAD_PLUGINS` is triggered. At one point `\craft\web\View::renderObjectTemplate` is called, which ends up initialising `CraftVariable`. Because  `Plugins::EVENT_AFTER_LOAD_PLUGINS` gets triggered before the modules are bootstrapped, I never get to catch the `CraftVariable` initialisation.

**Comment by khalwat** [comment]
Happy to make whatever changes make sense. This is something I struggled with as well, where other plugins would be doing something with Twig, and it caused all sorts of havoc. That's why I do the bulk of my initialization in `Plugins::EVENT_AFTER_LOAD_PLUGINS`

**Comment by brandonkelly** [comment]
As of the next release, plugins will be loaded before modules once again, but in a way that maintains the fix for #2761.

Since there’s no good workaround for this bug, we may release a 3.0.5.1 later today.

**Comment by khalwat** [comment]
Looks like whatever was changed here also broke the SEOmatic rendering process :/

https://github.com/nystudio107/craft-seomatic/issues/101
